### PR TITLE
Openapi 0.9.0

### PR DESCRIPTION
**Pull request by laura-rodriguez**
Add Application Integration Tests - Work in progress

### CODE REVIEW

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task AddSwaCustomApp()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new AutoLoginApplication
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Helper layer: Add `CreateApplicationAsync(CreateSwaCustomApplicationOptions)` overload

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task AddSamlCustomApp()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new SamlApplication
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Helper layer: Add `CreateApplicationAsync(CreateSamlApplicationOptions)` overload

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+            {
+                Label = "Sample SAML Custom App",
+                SignOnMode = ApplicationSignOnMode.Saml20,
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: See if we can use errata to rename this to `ApplicationSignOnMode.Saml2`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+                        Recipient = "http://testorgone.okta",
+                        Destination = "http://testorgone.okta",
+                        SubjectNameIdTemplate = "${user.userName}",
+                        SubjectNameIdFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+                        ResponseSigned = true,
+                        AssertionSigned = true,
+                        SignatureAlgorithm = "RSA_SHA256",
+                        DigestAlgorithm = "SHA256",
+                        HonorForceAuthn = true,
+                        AuthnContextClassRef = "urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport",
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename: `AuthenticationContextClassName`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+                        Audience = "asdqwe123",
+                        Recipient = "http://testorgone.okta",
+                        Destination = "http://testorgone.okta",
+                        SubjectNameIdTemplate = "${user.userName}",
+                        SubjectNameIdFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+                        ResponseSigned = true,
+                        AssertionSigned = true,
+                        SignatureAlgorithm = "RSA_SHA256",
+                        DigestAlgorithm = "SHA256",
+                        HonorForceAuthn = true,
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename: `HonorForceAuthentication`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+                    {
+                        RequestIntegration = false,
+                        Url = "https://example.com/bookmark.htm",
+                    },
+                },
+            });
+
+            try
+            {
+                var retrieved = await client.Applications.GetApplicationAsync(createdApp.Id);
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Discuss adding a helper: `client.Applications.GetApplicationAsync<IApplicationType>(appId)` when the application type is already known (like here). This would remove the need for casts in these tests.

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        /// <summary>
+        /// FAIL: GroupName expected to be null or empty but is not
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: It sounds like these are bugs with our example code on developer.okta.com. In that case, file it as an issue on https://github.com/okta/okta.github.io and change the values here to get the test working. 👍

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 8]
```diff
@@ -0,0 +1,553 @@
...
+        /// FAIL: wReplyOverride expected to be https://example.com but it came with a final `/`
+        /// </summary>
+        [Fact]
+        public async Task AddWsFederationApp()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new WsFederationApplication
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Helper layer: Add `CreateApplicationAsync(CreateWsFederationApplicationOptions)` overload

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 9]
```diff
@@ -0,0 +1,553 @@
...
+                    App = new WsFederationApplicationSettingsApplication()
+                    {
+                        AudienceRestriction = "urn:example:app",
+                        GroupName = null,
+                        GroupValueFormat = "windowsDomainQualifiedName",
+                        Realm = "urn:example:app",
+                        WReplyUrl = "https://example.com/",
+                        AttributeStatements = null,
+                        NameIdFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+                        AuthnContextClassRef = "urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport",
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename: `AuthenticationContextClassName`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 10]
```diff
@@ -0,0 +1,553 @@
...
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        /// <summary>
+        /// FAIL: Invalid grantTypes & responseTypes
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: We can discuss this 👍

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 11]
```diff
@@ -0,0 +1,553 @@
...
+        /// </summary>
+        [Fact]
+        public async Task AddOAuthTwoApp()
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Rename to: `AddOpenIdConnectApp`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 12]
```diff
@@ -0,0 +1,553 @@
...
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new OpenIdConnectApplication
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Helper layer: Add `CreateApplicationAsync(CreateOpenIdConnectApplication)` overload

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 13]
```diff
@@ -0,0 +1,553 @@
...
+            {
+                Name = "oidc_client",
+                Label = "Sample Client",
+                SignOnMode = ApplicationSignOnMode.OpenidConnect,
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename to: `ApplicationSignOnMode.OpenIdConnect`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 14]
```diff
@@ -0,0 +1,553 @@
...
+                {
+                    OauthClient = new ApplicationCredentialsOAuthClient()
+                    {
+                        ClientId = "0oa1hm4POxgJM6CPu0g4",
+                        TokenEndpointAuthMethod = OAuthEndpointAuthenticationMethod.ClientSecretPost,
+                        AutoKeyRotation = true,
+                    },
+                },
+
+                // MISSING PROPERTY: "post_logout_redirect_uris": [ "https://example.com/oauth2/postLogoutRedirectUri"],
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Reminder to discuss ⬆️

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 15]
```diff
@@ -0,0 +1,553 @@
...
+                             "https://example.com/oauth2/callback",
+                             "myapp://callback",
+                        },
+                        GrantTypes = new List<OAuthGrantType>
+                        {
+                            OAuthGrantType.Implicit,
+                            OAuthGrantType.AuthorizationCode,
+                        },
+                        ApplicationType = OpenIdConnectApplicationType.Native,
+                        TosUri = "https://example.com/client/tos",
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename: `TermsOfServiceUri`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 16]
```diff
@@ -0,0 +1,553 @@
...
+            });
+
+            try
+            {
+                var retrieved = await client.Applications.GetApplicationAsync(createdApp.Id);
+                retrieved.Name.Should().Be("oidc_client");
+                retrieved.Label.Should().Be("Sample Client");
+                retrieved.SignOnMode.Should().Be(ApplicationSignOnMode.OpenidConnect);
+
+                var retrievedCredentials = retrieved.GetProperty<ApplicationCredentialsOAuthClient>("credentials");
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Does `retrieved.Credentials` not work?

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 17]
```diff
@@ -0,0 +1,553 @@
...
+
+                retrievedCredentials.ClientId.Should().Be("0oa1hm4POxgJM6CPu0g4");
+                retrievedCredentials.AutoKeyRotation.Should().BeTrue();
+                retrievedCredentials.TokenEndpointAuthMethod.Should().Be(OAuthEndpointAuthenticationMethod.ClientSecretPost);
+
+                var retrievedSettings = retrieved.GetProperty<OpenIdConnectApplicationSettings>("settings");
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Does `retrieved.settings` not work?

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 18]
```diff
@@ -0,0 +1,553 @@
...
+
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ClientUri.Should().Be("https://example.com/client");
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Use errata to rename property to: `OAuthClient`

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 19]
```diff
@@ -0,0 +1,803 @@
...
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task ListApplications()
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Add another test very similar to this one: `ListApplicationsOfType`

* Add two applications (of different types)
* Make sure LINQ `OfType` works, like this: https://github.com/okta/okta-sdk-dotnet/blob/master/src/Okta.Sdk.IntegrationTests/FactorScenarios.cs#L50

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 20]
```diff
@@ -0,0 +1,803 @@
...
+                    Scheme = ApplicationCredentialsScheme.AdminSetsCredentials,
+                    UserNameTemplate = new ApplicationCredentialsUsernameTemplate()
+                    {
+                        Template = "${source.login}",
+                        Type = "BUILT_IN",
+                    },
+                };
+
+                retrieved.Credentials = schemeAppCredentials;
+                retrieved = await client.Applications.UpdateApplicationAsync(retrieved, retrieved.Id);
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Does `retrieved.UpdateAsync()` work here?

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 21]
```diff
@@ -0,0 +1,803 @@
...
+
+                // Checking defaults
+                var retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.EditUsernameAndPassword);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+
+                var schemeAppCredentials = new SchemeApplicationCredentials()
+                {
+                    Scheme = ApplicationCredentialsScheme.EditPasswordOnly,
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: Does this work if you change the original `retrievedCredentials` object, instead of creating a new `SchemeApplicationCredentials`?

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs (hunk 6 above)
Comment (nbarbettini) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: In other words, I want to be able to do:

```csharp
var retrieved = await client.Applications.GetApplicationAsync<IBookmarkApplication>(id);
```
and have `retrieved` be typed as `IBookmarkApplication`.

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs (hunk 10 above)
Comment (nbarbettini) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: It looks like the SDK is sending this:

```
...
"response_types": [
    {"value":"token"}, {"value":"id_token"}, {"value":"code"}
]
```

when it should be sending this:

```
...
"response_types": [
    "token", "id_token", "code"
]
```

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 22]
```diff
@@ -0,0 +1,803 @@
...
+                retrieved.Credentials = schemeAppCredentials;
+                retrieved = await client.Applications.UpdateApplicationAsync(retrieved, retrieved.Id);
+
+                retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.SharedUsernameAndPassword);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+                retrievedCredentials.UserName.Should().Be("sharedusername");
+
+                // Fail - Password is null
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: This is expected - Okta does not send back credentials (they can only be set). The test is probably working fine 🙂 you can confirm with Fiddler.

File: src/Okta.Sdk/ApplicationsClient.cs [hunk 23]
```diff
@@ -0,0 +1,280 @@
...
+                        Url = basicAuthApplicationOptions.Url,
+                        AuthUrl = basicAuthApplicationOptions.AuthUrl,
+                    },
+                },
+            };
+
+            return CreateApplicationAsync(app, activate, cancellationToken);
+        }
+
+        public Task<IApplication> CreateApplicationAsync(CreateBookmarkApplicationOptions bookmarkApplicationOptions, bool? activate = true, CancellationToken cancellationToken = default(CancellationToken))
```
[the remark this answers is not in the document]
Comment (nbarbettini) [reply src/Okta.Sdk/ApplicationsClient.cs]: (The same for all of these generic overloads)

[the remark this answers is not in the document]
Comment (nbarbettini) [reply src/Okta.Sdk/ApplicationsClient.cs]: Also, the `activate` parameter should be moved out of the method signatures and into each POCO.

File: src/Okta.Sdk/IOktaClient.cs [hunk 24]
```diff
@@ -33,6 +33,14 @@ public interface IOktaClient
         /// </value>
         IUsersClient Users { get; }
 
+        /// <summary>
+        /// Gets a <see cref="IApplicationsClient">ApplicationsClient</see> that interacts with the Okta Applications API.
+        /// </summary>i
```
Comment (nbarbettini) [inline src/Okta.Sdk/IOktaClient.cs]: typo?

File: src/Okta.Sdk/Internal/AbstractResourceTypeResolver.cs [hunk 25]
```diff
@@ -5,21 +5,34 @@
...
+        public Type GetResolvedType(IDictionary<string, object> data)
+        {
+            var resourceType = GetResolvedTypeInternal(data);
+
+            if (ResourceTypeResolverFactory.GetResolver(resourceType) == null)
+            {
+                return resourceType;
+            }
+
+            var resolverType = ResourceTypeResolverFactory.GetResolver(resourceType);
```
Comment (nbarbettini) [inline src/Okta.Sdk/Internal/AbstractResourceTypeResolver.cs]: Add comment:

// If there is a more specific resolver available, resolve again recursively

File: src/Okta.Sdk/Internal/CastingListAdapter{T}.cs [hunk 26]
```diff
@@ -45,6 +45,16 @@ private T Cast(object item)
                 return _resourceFactory.CreateFromExistingData<T>(nestedData);
             }
 
+            /***** Check if this logic should be encapsulated in another place *****/
```
Comment (nbarbettini) [inline src/Okta.Sdk/Internal/CastingListAdapter{T}.cs]: (Reminder: Remove these comments when this refactoring is done)

File: src/Okta.Sdk/Internal/CastingListAdapter{T}.cs [hunk 27]
```diff
@@ -45,6 +45,16 @@ private T Cast(object item)
...
+            var isStringEnum = StringEnum.StringEnumTypeInfo.IsAssignableFrom(_targetTypeInfo);
+
+            if (isStringEnum)
+            {
+                return (T)Activator.CreateInstance(typeof(T), item);
```
Comment (nbarbettini) [inline src/Okta.Sdk/Internal/CastingListAdapter{T}.cs]: We decided to move this to a static method on `StringEnum` right? (I think so)

File: src/Okta.Sdk/Internal/ResourceTypeResolverFactory.cs [hunk 28]
```diff
@@ -31,16 +31,24 @@ internal static class ResourceTypeResolver
...
                 return new DefaultResourceTypeResolver<T>();
             }
 
-            return (AbstractResourceTypeResolver<T>)Activator.CreateInstance(resolver);
+            return (AbstractResourceTypeResolver<T>)Activator.CreateInstance(resolverType);
+        }
+
+        public static IResourceTypeResolver Create(Type resourceType)
+        {
+            return (IResourceTypeResolver)Activator.CreateInstance(resourceType);
```
Comment (nbarbettini) [inline src/Okta.Sdk/Internal/ResourceTypeResolverFactory.cs]: Is this doing the same thing as line 46?

File: src/Okta.Sdk/StringEnum.cs [hunk 29]
```diff
@@ -12,6 +15,8 @@ namespace Okta.Sdk
     /// </summary>
     public abstract class StringEnum : IComparable
     {
+        internal static readonly TypeInfo StringEnumTypeInfo = typeof(StringEnum).GetTypeInfo();
```
Comment (nbarbettini) [inline src/Okta.Sdk/StringEnum.cs]: Rename to `TypeInfo`

File: src/Okta.Sdk/Internal/StringEnumSerializingConverter.cs [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Okta.Sdk.Internal
+{
+    public sealed class StringEnumSerializingConverter : JsonConverter
+    {
+        private static readonly TypeInfo StringEnumTypeInfo = typeof(StringEnum).GetTypeInfo();
```
Comment (nbarbettini) [inline src/Okta.Sdk/Internal/StringEnumSerializingConverter.cs]: You added this to `StringEnum` already, so no need to add it here again.

File: src/Okta.Sdk/Resource.cs [hunk 31]
```diff
@@ -98,7 +97,7 @@ public T GetProperty<T>(string name)
                 return GetResourcePropertyInternal<T>(name);
             }
 
-            if (StringEnumTypeInfo.IsAssignableFrom(typeInfo))
+            if (typeof(T) == typeof(StringEnum))
```
Comment (nbarbettini) [inline src/Okta.Sdk/Resource.cs]: Why this change? (seems like it would exclude classes that inherit from `StringEnum`)

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs (hunk 7 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: https://github.com/okta/okta.github.io/issues/2141

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs (hunk 14 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: https://github.com/okta/okta-sdk-dotnet/issues/209

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 32]
```diff
@@ -0,0 +1,632 @@
...
+                SsoAcsUrl = "http://testorgone.okta",
+                IdpIssuer = "http://www.okta.com/${org.externalKey}",
+                Audience = "asdqwe123",
+                Recipient = "http://testorgone.okta",
+                Destination = "http://testorgone.okta",
+                SubjectNameIdTemplate = "${user.userName}",
+                SubjectNameIdFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+                ResponseSigned = true,
+                AssertionSigned = true,
+                // Enum here?
```
[the remark this answers is not in the document]
Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs]: https://github.com/okta/okta-sdk-dotnet/issues/211

File: src/Okta.Sdk/Internal/ResourceTypeResolverFactory.cs (hunk 28 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk/Internal/ResourceTypeResolverFactory.cs]: Not exactly the same, here we cast to the new interface `IResourceTypeResolver`. The one from line 46 is returning objects cast as the abstract class.

I kept both, but we can opt to use only that returns `IResourceTypeResolver`

File: src/Okta.Sdk/StringEnum.cs [hunk 33]
```diff
@@ -96,5 +102,21 @@ public override bool Equals(object obj)
         /// <param name="other">The object to compare to.</param>
         public int CompareTo(object other)
             => string.Compare(Value, ((StringEnum)other).Value, StringComparison.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Creates a new <see cref="StringEnum"/> from an existing dictionary.
+        /// </summary>
+        /// <typeparam name="T">The <see cref="StringEnum"/> type.</typeparam>
+        /// <param name="item">The existing dictionary.</param>
```
[the remark this answers is not in the document]
Comment (nbarbettini) [reply src/Okta.Sdk/StringEnum.cs]: Note: I fixed these in my latest commit.

File: openapi/errata.js [hunk 34]
```diff
@@ -1,11 +1,10 @@
 const propertyErrata = [
   { path: 'FactorDevice.links', skip: true, skipReason: 'Not currently supported' },
   { path: 'Link.hints', skip: true, skipReason: 'Not currently supported' },
-  { path: 'User._links', skip: true, skipReason: 'Not currently supported' },
-  { path: 'UserGroup._embedded', skip: true, skipReason: 'Not currently supported' },
-  { path: 'UserGroup._links', skip: true, skipReason: 'Not currently supported' },
-  { path: 'UserGroupStats._links', skip: true, skipReason: 'Not currently supported' },
   
+  { path: '*._embedded', skip: true, skipReason: 'Not currently supported', apply: 'All' },
```
[the remark this answers is not in the document]
Comment (laura-rodriguez) [reply openapi/errata.js]: Agree
